tests/test_views_records.py

```diff
@@ -38,15 +38,19 @@ def test_unknown_bucket_raises_403(self):
 
     def test_unknown_collection_raises_404(self):
         other_collection = self.collection_url.replace('barley', 'pills')
-        self.app.get(other_collection, headers=self.headers, status=404)
+        resp = self.app.get(other_collection, headers=self.headers, status=404)
+        self.assertIn('id', resp.json['details'])
+        self.assertIn('resource_name', resp.json['details'])
```

Comment:
It looks like you're only checking that the attributes are present in the error message, not that they are correct. I think we can add stronger tests that also check that `resp.json['details']['id']` is equal to `"barley"`, and similarly that the `resource_name` is `"collection"`.

I'm not sure every single test needs to be updated. Personally, I think it's OK if we leave these tests alone -- they check that we got a 404, and that's good enough. Instead I would add new tests that check that when we make a request that we expect to get a 404, that that 404 comes back with these details.

Since we expect all 404s to have a certain structure (with `id` and `resource_name` having certain attributes), it might be worthwhile to add a method somewhere called something like `check404` that enforces this structure. Then, the tests that use this method will look more declarative.